app/modelmodule/admin_acls.py

```diff
@@ -0,0 +1,65 @@
+"""Model for admin/moderator access control."""
+
+
+from google.appengine.ext import db
+
+
+class AdminPermission(db.Model):
+    """Entity class for tracking admin/moderator access.
+
+    There should be only one AdminPermission entity for any given repo/email
+    address pair.
```

Comment:
in general, could there be an edge case where a user is eg a moderator until Jan 1, and later on they should be an administrator - but there's already a pair with their email and the repo, so they can't be added until the old entry is deleted?
Maybe that's working as intended - just wanted to flag it.

Reply:
I can't think of a situation where we'd need to support timed upgrades/downgrades, I think it'll be fine to require that someone just edit the existing entry.